mcu/stm32f0xx.S

```diff
@@ -274,15 +303,22 @@ System_Reset:
 
     .size Reset_Handler, . - Reset_Handler
 
-/*    Macro to define default handlers. Default handler
- *    will be weak symbol and just dead loops. They can be
- *    overwritten by other handlers */
+/*    Macro to define default handlers.
+ *    Since F0 doesn't support relocation of the ISR table
+ *    we need to resolve a new ISR vector via SCB->ICSR.
+ */
     .align 2
     .thumb_func
     .type _default_handler, %function
 
 _default_handler:
-    b .
+    ldr     r0, = SCB
+    ldr     r1, = _APP_START
+    ldr     r0, [r0, SCB_ICSR]
+    lsls    r0, 26
+    lsrs    r0, 24
+    ldr     r0, [r1, r0]
+    bx      r0
```

Comment:
Forgive my intrusion, but IMHO we need to optimize this to the bone.

I'm pretty sure we remove at least one instruction - by loading ICSR with absolute address

Reply:
What do you mean?
```
ldr r0, [pc + some offset] // loading absolute address of the SCB->ICSR from .pool
ldr r0, [r0 + 0x00]        // loading value of the SCB->ICSR register
```
is the same as
```
ldr r0, [pc + some offset] // loading absolute address of the SCB from .pool
ldr r0, [r0 + SCB_ICSR]    // loading value of the SCB->ICSR register
```
But we already have an address of the SCB in the ```.pool```.

Reply:
There are no operations with 32-bit direct addressing for the Cortex-M0

Reply:
I thought that you could do

```
#define SCB_ICSR_ABS (SCB + SCB_ICSR)
...
ldr r0 = SCB_ICSR_ABS
```

Reply:
I see, that does not solve anything because dereferencing does the addition for free.